stonesoup/types/array.py

```diff
@@ -29,6 +29,38 @@ def __rmatmul__(self, other):
         out = np.matmul(np.asfarray(other), np.asfarray(self))
         return out.view(type=type(other))
 
+    @classmethod
+    def _cast(cls, val):
+        # This tries to cast the result as either a StateVector or
+        # CovarianceMatrix if applicable.
+        if val.ndim == 1 and val.shape == (1,):
```

Comment:
Think this if statement is redundant given _else_ below.

Reply:
I updated the _cast() function.